Show the number of new messages next to "Inbox" in the nav bar

alice-blue opened the pull request:
This displays the number of new messages next to the word "Inbox" in the nav bar at the top of the page.

I experimented a little with trying to make the number stand out more by using red type, or white type on a red background, but I didn't find a combo that looked good to me (especially in dark mode), so this PR addresses only part of https://github.com/iftechfoundation/ifdb/issues/1213

![inbox messages](https://github.com/user-attachments/assets/9b8e7d93-b3ef-43be-af3f-0e5dea00a3a2)

(Originally I was testing this by logging in as different users, but I figured out that it also works if you comment on your own profile.)

salty-horse commented on www/pagetpl.php:
```diff
@@ -113,7 +114,22 @@ function pageHeader($title, $focusCtl = false, $extraOnLoad = false,
+                    $uid = checkPersistentLogin();
+                    $quid = mysql_real_escape_string($uid, $db);
+                    $inboxCnt = 0;
+                    if ($quid) {
```
check `$uid` here, and move `mysql_real_escape_string` inside the `if`.

(Later I will change it so queryComments gets `$uid` as an argument)

alice-blue replied:
Ok, I moved that and also moved $db inside the if.

salty-horse commented on www/pagetpl.php:
```diff
@@ -113,7 +114,22 @@ function pageHeader($title, $focusCtl = false, $extraOnLoad = false,
+                    $quid = mysql_real_escape_string($uid, $db);
+                    $inboxCnt = 0;
+                    if ($quid) {
+                        list($inbox, $inboxCnt) =
```
I prefer `[]` instead of `list()` in new code.

alice-blue replied:
Changed this.

dfabulich commented on www/pagetpl.php:
```diff
@@ -113,7 +114,22 @@ function pageHeader($title, $focusCtl = false, $extraOnLoad = false,
+                    <?php
+                    // check the inbox so we can display the number of new messages
+                    $inboxCnt = 0;
+                    $uid = checkPersistentLogin();
```
We already called `checkPersistentLogin()` at line 82. We probably don't want to do it again here.

alice-blue replied:
I changed references to $uid to say $curuser instead

dfabulich commented on www/pagetpl.php:
```diff
@@ -113,7 +114,22 @@ function pageHeader($title, $focusCtl = false, $extraOnLoad = false,
+                    // check the inbox so we can display the number of new messages
+                    $inboxCnt = 0;
+                    $uid = checkPersistentLogin();
+                    if ($uid) {
```
I think we actually don't need an `if` statement here, because we checked if we're logged in on line 113.

alice-blue replied:
Ok, I removed this.

dfabulich commented on www/pagetpl.php:
```diff
@@ -113,7 +114,22 @@ function pageHeader($title, $focusCtl = false, $extraOnLoad = false,
+                        $db = dbConnect();
+                        $quid = mysql_real_escape_string($uid, $db);
+                        [$inbox, $inboxCnt] =
+                        queryComments($db, "inbox", $quid, "limit 0, 1", $caughtUpDate, false);
```
This line isn't working right. I see this warning in the Docker console:

```
PHP Notice:  Undefined variable: caughtUpDate in /var/www/html/pagetpl.php on line 125
```

And, sure enough, the number flagged in the inbox is the count of all comments for all time; even if you go to the inbox and click "I'm caught up," the Inbox header number doesn't decrease or go away.

`home` sets `$caughtUpDate` on line 27. I think we'd need to do something similar here in `pagetpl.php`.

alice-blue replied:
Ok, $caughtUpDate is now set.

dfabulich reviewed (changes requested):
We've got to fix the `caughtUpDate` before we should merge this.

dfabulich commented:
It would also be nice if we could skip calling `queryComments` in `check-inbox.php`, but that seems kinda tricky. (Maybe use a `global`?)

salty-horse commented:
See also #917

dfabulich commented:
I filed https://github.com/alice-blue/ifdb/pull/7 to show the Inbox count in a red number.

salty-horse commented:
If the number gets too big, it would look bad in a circle. Maybe it should cap at "99+".

alice-blue commented:
> It would also be nice if we could skip calling `queryComments` in `check-inbox.php`, but that seems kinda tricky. (Maybe use a `global`?)

@dfabulich I wasn't really sure how to do this, but tried anyway. I deleted the call to queryComments from check-inbox, and also added globals (I think?) in commentutil.php.

dfabulich reviewed (changes requested):
Minor conflict with #1217

alice-blue commented:
> Minor conflict with #1217

I think I unconflicted it.
